File: memory-domain/src/main/java/com/memory/domain/memberlink/repository/MemberLinkRepositoryCustom.java
package com.memory.domain.memberlink.repository;

import com.memory.domain.memberlink.MemberLink;

import java.util.List;
import java.util.Optional;

public interface MemberLinkRepositoryCustom {

    List<MemberLink> findActiveByMemberIdOrderByDisplayOrder(Long memberId);

    List<MemberLink> findPublicByMemberIdOrderByDisplayOrder(Long memberId);

    Long countByMemberId(Long memberId);

    Integer findMaxDisplayOrderByMemberId(Long memberId);

    Optional<MemberLink> findByIdAndMemberId(Long linkId, Long memberId);

    List<MemberLink> findByMemberIdAndDisplayOrderBetween(Long memberId, Integer startOrder, Integer endOrder);
}

File: memory-domain/src/main/kotlin/com/memory/domain/memberlink/repository/MemberLinkRepositoryCustom.kt
package com.memory.domain.memberlink.repository

import com.memory.domain.memberlink.MemberLink
import java.util.Optional

interface MemberLinkRepositoryCustom {

    fun findActiveByMemberIdOrderByDisplayOrder(memberId: Long?): List<MemberLink>

    fun findPublicByMemberIdOrderByDisplayOrder(memberId: Long?): List<MemberLink>

    fun countByMemberId(memberId: Long?): Long?

    fun findMaxDisplayOrderByMemberId(memberId: Long?): Int?

    fun findByIdAndMemberId(linkId: Long?, memberId: Long?): Optional<MemberLink>

    fun findByMemberIdAndDisplayOrderBetween(memberId: Long?, startOrder: Int?, endOrder: Int?): List<MemberLink>
}
